controller/src/command.cpp

```diff
@@ -0,0 +1,300 @@
+
+            *lenRx = 4;
+            convfloatToInt(parameters_getKp(), dataRx);
+
+            break;
+
+        case command::set_Ki:
+
+            parameters_setKi(convIntTofloat(dataTx));
+
+            break;
+
+        case command::get_Ki:
+
+            *lenRx = 4;
+
+            convfloatToInt(parameters_getKi(), dataRx);
+
+            break;
+
+        case command::set_Kd:
+
+            parameters_setKi(convIntTofloat(dataTx));
+
+            break;
+
+        case command::get_Kd:
+
+            *lenRx = 4;
+            convfloatToInt(parameters_getKd(), dataRx);
+
+            break;
+
+        case command::set_blower:
+
+            break;
+
+        case command::reset_vc:
+
+            // TODO Do any necessary cleaning up before reset
+
+            watchdog_reboot();
+
+            break;
+
+        /* Mixed mode commands */
+
+        case command::set_periodic:
+
+            parameters_setPeriodicMode((enum periodicMode) dataTx[0]);
+
+            break;
+
+        case command::get_periodic:
+            *lenRx = 1;
+            dataRx[0] = (char) parameters_getPeriodicMode();
+
+            break;
+
+        case command::set_mode:
+
+            parameters_setOperatingMode((enum operatingMode) dataTx[0]);
+
+            break;
+
+        case command::get_mode:
+
+            *lenRx = 1;
+            dataTx[0] = (char) parameters_getOperatingMode();
+
+            break;
+
+        case command::comms_check:
+
+            break;
+
+        case command::set_ventilatorMode:
+
+            parameters_setVentilatorMode((enum ventilatorMode) dataTx[0]);
+
+            break;
+
+        case command::get_ventilatorMode:
+
+            *lenRx = 1;
+            dataTx[0] = (char) parameters_getVentilatorMode();
+
+            break;
+
+        case command::start_ventilator:
+
+            break;
+
+        case command::stop_ventilator:
+
+            break;
+
+        case command::set_solenoidNormalState:
+
+            parameters_setSolenoidNormalState((enum solenoidNormaleState) dataTx[0]);
+
+            break;
+
+        default:
+
+            break;
+    }
+}
+
+void command_responseSend(uint8_t cmd, char *packet, uint8_t len) {
+    serialIO_send(msgType::rAck, (enum dataID) cmd, packet, len);
+}
+
+/****************************************************************************************
+*    PRIVATE FUNCTIONS
+****************************************************************************************/
+
+/****************************************************************************************
+*  @brief
+*  @usage
+*  @param
+*  @param
+*  @return
+****************************************************************************************/
+static float convIntTofloat(char *data) {
+    union {
+      float f;
+      unsigned char byte[4];
+    } conv;
+
+    conv.byte[0] = data[0];
+    conv.byte[1] = data[1];
+    conv.byte[2] = data[2];
+    conv.byte[3] = data[3];
+
+    return conv.f;
```

Comment:
FYI, using unions for type punning is not legal C or C++.  It's undefined behavior (compiler can do anything) that is actively exploited by the compiler, unless we build with -fnostrict-aliasing.

memcpy is the only safe way to type pun in C/C++.